### DISCUSSION TITLE
Transitionend event listener doesn't work on Safari (macOS + iOS)

### DISCUSSION THREAD

**Discussion by kslstn** [alpinejs/alpine#3288] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2810, where [SimoTod suggests](https://github.com/alpinejs/alpine/discussions/2810#discussioncomment-2530328) that increasing the `delay` in `requestAnimationFrame()` would solve the issue, which seems to work.

<div type='discussions-op-text'>

<sup>Originally posted by **kslstn** April  7, 2022</sup>
Safari ignores the expression for `x-on:transitionend` when used like this:
```html
<p
  x-show="show"
  x-cloak
  x-transition:enter="transition"
  x-transition:enter-start="opacity-0"
  x-transition:enter-end="opactity-1"
  x-transition:leave="transition"		 
  x-transition:leave-start="opacity-1"
  x-transition:leave-end="opacity-0"
  x-on:transitionend="transitionCount++"
></p>
```
```css
.transition{
  transition: all linear 200ms;
}
.opacity-0{
  opacity: 0;
}
.opacity-1{
  opacity: 1;
}
```
See [this Codepen for a demo](https://codepen.io/kslstn/pen/KKZQxRM).

Firefox and Chrome do detect the event. Safari does react to `ontransitionend` in [this plain JS example](https://codepen.io/kslstn/pen/wvpyOpX). And it also reacts to `x-on:transitionstart`.</div>

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
